ckanext/ozwillo_theme/fanstatic/theme_location.js

```diff
@@ -0,0 +1,77 @@
+"use strict";
+
+/* theme_location
+ *
+ * This JavaScript module gets a geojson from osm given a location name
+ * TODO: Find how to manage properly the required field,
+ *       Manage the case when the user click nowhere else than validate after typing the location
+ *       For now, the request always fails, find why
+ *       Find how to manage errors popup language
+ *
+ */
+
+ckan.module('theme_location', function ($) {
+  return {
+    initialize : function () {
+      var val = document.getElementById("field-spatial-name");
+      val.onchange = function(){nominatim(val.value)};
+      var myform = document.getElementById("dataset-edit");
+      myform.onsubmit = function(){return validateForm()};
+      var element = document.getElementById('custom_fields');
+      element.firstElementChild.firstElementChild.style.display = 'none';
+
+      $( function() {
+        $( "#field-spatial-name" ).autocomplete({
+          minLength: 2,
+          source: function (request, response) {
+            var query = request.term;
+            var url = 'https://search.osmnames.org/fr/q/' + query + '.js?key=xS5b3xnPgEUW5zi0GvWa';
```

Comment:
what is this key ? could it be externalized ?

Reply:
It is the API key for osmnames, yes I will add it to the configuration file